src/udp_wrap.cc

```diff
@@ -446,6 +446,10 @@ void UDPWrap::OnRecv(uv_udp_t* handle,
   }
 
   char* base = static_cast<char*>(realloc(buf->base, nread));
+  // If realloc fails, fallback to old pointer
+  if(base == NULL && nread != 0)
+      base = buf->base;
```

Comment:
@sam-github in this case realloc is shrinking the buffer, so using the old larger buffer is ok.  And while ive looked at some implementations of realloc, and they all seem to just return the original pointer if a shrinking realloc fails. However this isn't actually in the spec, and in theory there could be an implementation that does not
@socketpair  C90 does specify that memory will be freed, although C99 seems to have modified it. However you are right, I can't think why just freeing if ncount == 0 wouldnt make sense